services/brig/src/Brig/User/Auth.hs

```diff
@@ -181,3 +183,14 @@ validateTokens ut at = do
                 unless (e == ZAuth.Expired) (throwE e)
     ck <- lift (lookupCookie ut) >>= maybe (throwE ZAuth.Invalid) return
     return (ZAuth.userTokenOf ut, ck)
+
+-- | Allow to login as any user without having the credentials.
+backdoorLogin :: BackdoorLogin -> CookieType -> ExceptT LoginError AppIO Access
+backdoorLogin (BackdoorLogin uid label) typ = do
+    let pw = PlainTextPassword ""
+    Data.authenticate uid pw `catchE` \case
```

Comment:
How about instead using `Data.reauthenticate` with `Nothing` as the password and catching `ReAuthMissingPassword` instead?

Reply:
why?  to me this makes semantically less sense by far.

Reply:
It makes less sense semantically but doesn't require "making up" a password. I can't say that I like either choice much

Reply:
(i'm ok with this change.  it's not an improvement in my opinion, but it's still perfectly readable and correct.)

Reply:
Maybe the rightest thing is to change `authenticate` to accept `Nothing`

Reply:
I think that if we want to be explicit, it'd be better to have a different function altogether and call it `authenticateInternal` or something like that, that explicitly does not check for anything else other than the checks that we want to do; I just think that passing an empty string as a password felt off